core/src/main/scala/chisel3/experimental/prefix.scala

```diff
@@ -55,6 +55,14 @@ object prefix {
     if (Builder.getPrefix.nonEmpty) Builder.popPrefix()
     ret
   }
+
+  def apply[T](name: Any)(f: => T): T = {
+    name match {
+      case x: HasId  => applyHasId[T](x)(f)
+      case x: String => applyString[T](x)(f)
+      case _ => f
+    }
+  }
```

Comment:
This change is fine for development, but I do not want to merge this. It is less type safe and strictly inferior to just overloading apply with the correct concrete subtypes.

If we must have something temporarily, we can have `applyHasId` and `applyString` _in addition to_ the overloaded `apply` methods, but there must be a way to make the Scala 3 plugin do the right thing.

Reply:
yeah it's not great - i'll move these into their own methods. it's on my todo list to ask around on how to get types resolved in the `appliedTo` calls